lib/high_voltage/page_collector.rb

```diff
@@ -0,0 +1,25 @@
+module HighVoltage
+  require "find"
+
+  class PageCollector
+    attr_reader :content_path
+
+    def initialize(content_path)
+      @content_path = content_path.to_s
+    end
+
+    def page_ids
+      page_ids = []
+
+      Find.find(content_path) do |file_path|
```

Comment:
This method is getting a bit long and seems to be responsible for quite a bit. What do you think about extracting a private method for collecting Page objects and then using a select on that collection for valid ones.

Something like:

```ruby
def page_ids
  pages.select(&:valid?).map(&:id)
end

private

def pages
  Find.find(content_path).map { |file_path| HighVoltage::Page.new(content_path, file_path) }
end
```